Sources/SourceControl/GitRepository.swift

```diff
@@ -114,6 +114,11 @@ public struct GitRepositoryProvider: RepositoryProvider {
                          failureMessage: "Failed to clone repository \(repository.location)",
                          progress: progressHandler)
     }
+
+    public func repositoryExists(at path: AbsolutePath) throws -> Bool {
+        // TODO: make this check more sophisticated?
```

Comment:
Maybe just also checking for `.git` subdirectory?  Is it correct to return false if there is a non-repository directory (or a file, etc) at `path`?  Almost feels like there are really there cases:  nothing there, repository there, or something else there.

Reply:
yup, we can also check for .git